lib/handle-filters.js

```diff
@@ -7,16 +7,21 @@ var error = require('pug-error');
 var runFilter = require('./run-filter');
 
 module.exports = handleFilters;
-function handleFilters(ast, filters) {
+function handleFilters(ast, filters, options) {
+  options = options || {}
   walk(ast, function (node) {
     var dir = node.filename ? dirname(node.filename) : null;
     if (node.type === 'Filter') {
       handleNestedFilters(node, filters);
       var text = getBodyAsText(node);
       var attrs = getAttributes(node);
-      attrs.filename = node.filename;
+      options = options[node.name] || {};
+      for (var attr in attrs) {
+        options[attr] = attrs[attr];
```

Comment:
Good point, latest commit should do the trick. I wanted to be sure that a user-specified option overrode the default option. The latest commit should keep that principle in tact.